assets/js/base/utils/price.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { sprintf } from '@wordpress/i18n';
+import { CURRENCY } from '@woocommerce/settings';
+
+/**
+ * Format a price with currency data.
+ *
+ * @param {number} value Number to format.
+ * @param {string} priceFormat  Price format string.
+ * @param {string} currencySymbol Curency symbol.
+ */
+export const formatPrice = (
+	value,
+	priceFormat = CURRENCY.price_format,
+	currencySymbol = CURRENCY.symbol
+) => {
+	if ( value === '' || undefined === value ) {
+		return '';
+	}
+	const formattedNumber = parseInt( value, 10 );
+	const formattedValue = sprintf(
+		priceFormat,
+		currencySymbol,
+		formattedNumber
+	);
+
+	// This uses a textarea to magically decode HTML currency symbols.
+	const txt = document.createElement( 'textarea' );
+	txt.innerHTML = formattedValue;
+	return txt.value;
```

Comment:
Are we sure this trick will handle all currency symbols/formats across various browsers?

Reply:
This code wasn't introduced in this PR, I just moved it into a general utils file for reuse. I believe the technique comes from https://stackoverflow.com/questions/7394748/whats-the-right-way-to-decode-a-string-that-has-special-html-entities-in-it/7394787#7394787.  I've spotted no issues so far in testing with woo core currencies across FF/Safari/Chrome.

Also not aware of a native solution for this. Since woo core uses entities I don't see an alternative unless you want to start storing all currency codes in blocks plugin :)